contract_line_tax/models/contract_line.py

```diff
@@ -19,6 +19,11 @@ def _prepare_invoice_line(self, move_form):
         self.ensure_one()
         res = super(ContractLine, self)._prepare_invoice_line(move_form)
         taxes = self.tax_ids or self.product_id.taxes_id
+        fiscal_position = self.contract_id.fiscal_position_id or self.env[
+            "account.fiscal.position"
+        ].get_fiscal_position(self.partner_id.id)
+        if taxes and fiscal_position:
+            taxes = fiscal_position.map_tax(taxes, partner=self.partner_id)
```

Comment:
I think it's clearer this way and avoid to look for the fiscal position if taxes are empty
```suggestion
        if taxes:
          fiscal_position = self.contract_id.fiscal_position_id or self.env[
              "account.fiscal.position"
          ].get_fiscal_position(self.partner_id.id)
          if fiscal_position:
              taxes = fiscal_position.map_tax(taxes, partner=self.partner_id)
```